### ISSUE TITLE
Exception didn't keep after converting from `Try` to `TryOption`

### ISSUE THREAD

**Issue by chakming** [louthy/language-ext#295]
Thanks for the great lib! 
tested with `2.1.18`
**Expected result**
when match, goes into `Fail` and `throw an exception with "some error"`

**Actual result**
treated as `None` and `throw new Exception("strange")`
```
public void CustomTest()
{
     var @flow =
           from a in A()
           from b in B(a).ToTryOption()
           select (a, b);
        
     @flow.Match(tuple => Unit.Default, () => throw new Exception("strange"), ex => throw ex);
}

public Try<int> B(int a)
{
      return () =>
      {
          throw new Exception("some error");
      };
}

public TryOption<int> A()
{
     return () =>
     {
         return 1;
     };
}
```

**Comment by chakming** [comment]
read the following source code, Is this expected behaviour? if so this seems a feature request 
https://github.com/louthy/language-ext/blob/master/LanguageExt.Core/DataTypes/Try/Try.Extensions.cs#L213-L215

**Comment by louthy** [comment]
@chakming Looks like a cut n paste error.  Fix committed, will roll out tonight hopefully.

**Comment by louthy** [comment]
Fixed in `v2.1.24` on nu-get now.
